### PR TITLE
💫 feat: change infrastructure condition field to ManyToMany field (#3970)

### PR DESCRIPTION
**Pull request by juggler31**
## Checklist

- [ ] I have followed the guidelines in our [Contributing document](https://geotrek.readthedocs.io/en/latest/contribute/contributing.html)
- [ ] My code respects the Definition of done available in the [Development section of the documentation](https://geotrek.readthedocs.io/en/latest/contribute/development.html#definition-of-done-for-new-model-fields)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes
- [x] I added an entry in the changelog file
- [x] My commits are all using prefix convention (emoji + tag name) and references associated issues
- [x] I added a label to the PR corresponding to the perimeter of my contribution
- [x] The title of my PR mentionned the issue associated

### CODE REVIEW

File: geotrek/infrastructure/tests/factories.py [hunk 1]
```diff
@@ -49,25 +49,46 @@ class Meta:
...
-    condition = factory.SubFactory(InfrastructureConditionFactory)
     published = True
+
+    @factory.post_generation
+    def conditions(obj, create, extracted=None, **kwargs):
+        if create:
+            if extracted:
+                obj.conditions.set(extracted)
+            else:
+                obj.conditions.add(InfrastructureConditionFactory.create())
```
Comment (submarcos) [inline geotrek/infrastructure/tests/factories.py:94]: can you factorize post_generation in an InfrastructureFatoyMixin ?

File: geotrek/infrastructure/migrations/0039_infrastructure_conditions.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+from django.db import migrations, models
+
+
+def copy_condition_to_conditions(apps, schema_editor):
+    Infrastructure = apps.get_model('infrastructure', 'Infrastructure')
+
+    # Copy InfrastructureCondition to SignageCondition and BladeCondition
+    for infrastructure in Infrastructure.objects.all():
+        infrastructure.conditions.add(infrastructure.condition)
```
Comment (Chatewgne) [inline geotrek/infrastructure/migrations/0039_infrastructure_conditions.py:11]: Fails if we have an infrastructure with no Condition

**Review by Chatewgne — CHANGES_REQUESTED** [review]

**Review by Chatewgne — APPROVED** [review]

File: geotrek/infrastructure/models.py [hunk 3]
```diff
@@ -213,6 +216,15 @@ def tourism_infrastructures(cls, tourism_obj, queryset=None):
     def outdoor_infrastructures(cls, outdoor_obj, queryset=None):
         return intersecting(qs=queryset_or_model(queryset, cls), obj=outdoor_obj)
 
+    @property
+    def conditions_display(self):
+        if hasattr(self, "conditions_list"):
+            # Use conditions prefetched
+            conditions_list = self.conditions_list
```
Comment (marcantoinedupre) [inline geotrek/infrastructure/models.py:223]: Je propose de ne pas utiliser l'attribut `conditions_list`. Il introduit une sorte de dépendance du modèle vers la view mais après test je ne constate pas d'amélioration des perfs en utilisant le param `to_attr` de `Prefetch` dans la View et le nb de requêtes SQL est le même (test avec 5000 aménagements).

Comment (submarcos) [reply geotrek/infrastructure/models.py:223]: dans ce cas il faut trouver un moyen de factoriser car il faut que le prefetch soit utilisé par les vues interne mapentity drf zet les exports csv shape etc...

Par contre je ne sais pas comment tu as testé mais le prefecth fonctionne tres bien chez moi

Comment (submarcos) [reply geotrek/infrastructure/models.py:223]: avec : 
image

sans :
image

File: docs/changelog.rst [hunk 4]
```diff
@@ -78,6 +81,7 @@ https://geotrek.readthedocs.io/en/latest/install/upgrade.html#postgresql
 - Bump mapentity to 8.9.0
 
 
+
```
Comment (submarcos) [inline docs/changelog.rst:84]:
```suggestion
```

File: docs/changelog.rst [hunk 5]
```diff
@@ -5,6 +5,9 @@ CHANGELOG
 2.108.0+dev     (XXXX-XX-XX)
 ----------------------------
 
+**Improvements**
+
+- Change infrastructure condition field to ManyToMany field (#3970)
```
Comment (submarcos) [inline docs/changelog.rst:11]:
```suggestion

```

File: docs/changelog.rst [hunk 6]
```diff
@@ -5,6 +5,9 @@ CHANGELOG
...
```
Comment (submarcos) [inline docs/changelog.rst:10]:
```suggestion
- Change infrastructure condition field to ManyToMany field (#3970)

```

File: docs/changelog.rst (hunk 4 above)
Comment (submarcos) [inline docs/changelog.rst:84]:
```suggestion
```

File: docs/changelog.rst [hunk 7]
```diff
@@ -129,7 +133,6 @@ https://geotrek.readthedocs.io/en/latest/install/upgrade.html#postgresql
 - Add structures to Aggregator (#3569)
 - Allow to filter flatpages by portal on admin list page
```
Comment (submarcos) [inline docs/changelog.rst:135]:
```suggestion


```

**Pull request merged**